src/styles/main.scss

```diff
@@ -1,3 +1,76 @@
-body {
+$days:
+  "mon" 0,
+  "tue" 101px,
+  "wed" 202px,
+  "thu" 303px,
+  "fri" 404px,
+  "sat" 505px,
+  "sun" 606px;
+$month-length: 28, 29, 30, 31;
+
+* {
+  box-sizing: border-box;
+}
+
+body,
+div {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  padding: 10px;
+  margin: 500px auto;
+  max-width: 800px;
+
+  &__day {
+    position: relative;
+
+    width: 100px;
+    height: 100px;
+
+    background-color: #eee;
+    border: 1px solid #000;
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: pink;
+
+      transform: translateY(-20px);
+      cursor: pointer;
+    }
+  }
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(#{$index})::before {
+      position: absolute;
+      top: 35px;
+      left: 35px;
+
+      content: "#{$index}";
+      color: #000;
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  @each $length in $month-length {
+    &--month-length-#{$length} {
+      @for $index from $length through 31 {
+        $temporary: $index + 1;
+        .calendar__day:nth-child(#{$temporary}) {
+          display: none;
+        }
+      }
+    }
+  }
```

Comment:
What does your each do?

Reply:
This @each loop takes turn-by-turn numbers of a $month-length variable. After that @each loop creates modifiers with one of $month-length number and the next loop @for starts with the current $length of month